### DISCUSSION TITLE
When data context is set in the control constructor, it overrides the design time data context

### DISCUSSION THREAD

**Discussion by DefinitelyADev** [AvaloniaUI/Avalonia#14583] [Q&A] [answered]
## Describe the bug
When setting the data context in the user control constructor, it overrides the design time data context.

## To Reproduce

1. In the provided solution, go to `UserControl1.axaml`
2. Build the solution
3. See the greeting set in the `Design.DataContext`
4. Go to `UserControl1.axaml.cs`
5. Uncomment line 12 with code `// DataContext = new UserControl1ViewModel();`
6. Go to `UserControl1.axaml`
7. Build the solution
8. See the greeting disappear

## Expected behavior
Show the values set in the `Design.DataContext` tag

## Screenshots

![image](https://github.com/AvaloniaUI/Avalonia/assets/20267678/4cc292b4-46a0-45fe-801e-260b57b47ac4)

![image](https://github.com/AvaloniaUI/Avalonia/assets/20267678/b1a89227-0124-4703-ae79-28edac14a4f6)

## Environment

- OS: Windows 11
- Avalonia-Version: 11.0.5 and 10.0.9

## Additional context
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/14230968/AvaloniaApplication1.zip)

**Comment by maxkatz6** [comment]
It's not a bug.
XAML property setters are executed inside of InitializeComponents() call.
So, you should be able to just move DataContext in the ctor to be executed before InitializeComponents.

**Reply by DefinitelyADev** [reply]
The same thing happens when I do this:
```cs
public UserControl1()
{
    DataContext = new UserControl1ViewModel();
    InitializeComponent();
}
```

**Reply by maxkatz6** [reply]
Ah, just checked. Yes, Design.DataContext is being set on your control another design control that hosts your content. But since your content redefines DataContext, host (design) context doesn't get inherited to the child.

**Reply by timunie** [reply, marked as answer]
if(Desing.IsDesingTime) can be used to skip that for the designer iirc.

**Reply by DefinitelyADev** [reply]
@timunie hmm this works, it is actually `Design.IsDesignMode`, though it is not as elegant as behind the scenes "magic". And it's a bit inconvenient to check everywhere but for now it's ok. Thanks!
